### PR TITLE
Add jobs for CAPI release 1.2

### PR DESCRIPTION
**Pull request by ykakarap**
Add jobs for Cluster API release v1.2

Cluster API issue: https://github.com/kubernetes-sigs/cluster-api/issues/6671

### CODE REVIEW

File: config/testgrids/kubernetes/sig-cluster-lifecycle/config.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ dashboard_groups:
     - sig-cluster-lifecycle-cluster-api-0.4
     - sig-cluster-lifecycle-cluster-api-1.0
     - sig-cluster-lifecycle-cluster-api-1.1
+    - sig-cluster-lifecycle-cluster-api-1.2
```
Comment (sbueringer) [inline config/testgrids/kubernetes/sig-cluster-lifecycle/config.yaml:12]: Also has to be added in ~ l.36

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-2.yaml [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    # The script this job runs is not in all branches.
+    - ^release-1.2$
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220614-cde7860a33-1.24
+        command:
+        - runner.sh
+        - ./scripts/ci-build.sh
+    annotations:
+      testgrid-dashboards: sig-cluster-lifecycle-cluster-api
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-2.yaml:19]: dashboard must be adjusted as well (please diff against the release-1.1 presubmits for that diff)

Same for other files

**Review by sbueringer — COMMENTED** [review]
I think otherwise it should be good, but I'll double-check after the fixes

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-2-upgrades.yaml [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+          - "./scripts/ci-e2e.sh"
+        env:
+          - name: KUBERNETES_VERSION_UPGRADE_FROM
+            value: "stable-1.24"
+          - name: KUBERNETES_VERSION_UPGRADE_TO
+            value: "ci/latest-1.25"
+          - name: ETCD_VERSION_UPGRADE_TO
+            value: "3.5.3-0"
+          - name: COREDNS_VERSION_UPGRADE_TO
+            value: "v1.8.6"
```
Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-2-upgrades.yaml:296]:
```suggestion
            value: "v1.9.3"
```

xref https://github.com/kubernetes-sigs/cluster-api/issues/6641

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-2.yaml [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+          - "./scripts/ci-e2e.sh"
+        env:
+          - name: KUBERNETES_VERSION_UPGRADE_FROM
+            value: "stable-1.24"
+          - name: KUBERNETES_VERSION_UPGRADE_TO
+            value: "ci/latest-1.25"
+          - name: ETCD_VERSION_UPGRADE_TO
+            value: "3.5.3-0"
+          - name: COREDNS_VERSION_UPGRADE_TO
+            value: "v1.8.6"
```
Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-2.yaml:266]:
```suggestion
            value: "v1.9.3"
```

xref https://github.com/kubernetes-sigs/cluster-api/issues/6641

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**